lib/core/include/irods/irods_logger.hpp

```diff
@@ -484,14 +485,49 @@ namespace irods::experimental::log
             ///
             /// \since 4.3.0
             template <typename... Args>
-            auto operator()(fmt::format_string<Args...> _format, Args&&... _args) const -> void
+            void operator()(fmt::format_string<Args...> _format, Args&&... _args) const
+                // Clang 15 and later (and GCC 11 and later) consider this overload ambiguous
+                // with the std::invocable overload. This constraint sorts that out.
```

Comment:
Let's move this inside the preprocessor conditional just like the one that follows.

Reply:
Done